WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -225,8 +225,9 @@ def post_results
         body += "#{record_strs.join(", ")}.  \n" # Trailing spaces for markdown give us a <br>
       end
     end
-    post = Post.new(title: title, body: body, author: current_user, world_readable: true)
-    post_post(post)
+    comp.update(results_posted_at: Time.now)
+    comp.save!
```

Comment:
Unfortunately, I fear that hiding results until they're ready won't be a trivial change to make, as we have a number of tables that are derived from the One True Results table (for example: [`ConciseAverageResults`]( https://github.com/jfly/worldcubeassociation.org/blob/efeaf4146c92d60b625a1ec502a14652c7a4b53d/WcaOnRails/db/structure.sql#L87)).

I'm unsure to what degree this is a theoretical problem. It may be that the window of incomplete results is so tiny that we just don't care about it. Would you like to start a discussion with the Results team to see how bad this window of time is in reality?